### PR TITLE
feat(auth): `revokeTokenWithAuthorizationCode()` implementation for revoking Apple sign-in token

### PR DESCRIPTION
**Pull request by russellwheatley**
## Description

Authorization code has been added to the `additionalUserInfo` returned on the `UserCredential` when signing in with apple on apple platforms which the user can pass as argument for `revokeTokenWithAuthorizationCode()`.

## Related Issues

closes https://github.com/firebase/flutterfire/issues/11065

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_auth/firebase_auth/lib/src/firebase_auth.dart [hunk 1]
```diff
@@ -821,6 +821,12 @@ class FirebaseAuth extends FirebasePluginPlatform {
     );
   }
 
+  /// Apple only. Users signed in with Apple provider can revoke their token using an authorization code.
+  /// Authorization code can retrieved on the user credential i.e. userCredential.additionalUserInfo.authorizationCode
```
Comment (russellwheatley) [inline packages/firebase_auth/firebase_auth/lib/src/firebase_auth.dart:825]:
```suggestion
  /// Authorization code can be retrieved on the user credential i.e. userCredential.additionalUserInfo.authorizationCode
```

File: packages/firebase_auth/firebase_auth_platform_interface/lib/src/platform_interface/platform_interface_firebase_auth.dart [hunk 2]
```diff
@@ -662,4 +662,11 @@ abstract class FirebaseAuthPlatform extends PlatformInterface {
   }) {
     throw UnimplementedError('verifyPhoneNumber() is not implemented');
   }
+
+  /// Apple only. Users signed in with Apple provider can revoke their token using an authorization code.
+  /// Authorization code can retrieved on the user credential i.e. userCredential.additionalUserInfo.authorizationCode
```
Comment (russellwheatley) [inline packages/firebase_auth/firebase_auth_platform_interface/lib/src/platform_interface/platform_interface_firebase_auth.dart:667]:
```suggestion
  /// Authorization code can be retrieved on the user credential i.e. userCredential.additionalUserInfo.authorizationCode
```

**Review by Lyokone — APPROVED** [review]

**Review by exaby73 — APPROVED** [review]

File: docs/auth/federated-auth.md [hunk 3]
```diff
@@ -232,6 +232,25 @@ Future<UserCredential> signInWithApple() async {
 }
 ```
 
+### Revoke Apple Auth Token
```
Comment (russellwheatley) [inline docs/auth/federated-auth.md:235]: @kevinthecheung - some more doc updates for you to review. Thanks!

Comment (kevinthecheung) [inline docs/auth/federated-auth.md:235]:
```suggestion
### Revoke Apple auth tokens
```

**Review by kevinthecheung — APPROVED** [review]

**Pull request merged**